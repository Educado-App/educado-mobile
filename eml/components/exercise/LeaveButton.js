import React from 'react';
import { StyleSheet } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { Icon, Button } from '@rneui/base';
import tailwindConfig from '../../tailwind.config';
const projectColors = tailwindConfig.theme.colors;
import PropTypes from 'prop-types';

const LeaveButton = ({ navigationPlace, courseId }) => {
	LeaveButton.propTypes = {
		navigationPlace: PropTypes.string.isRequired,
		courseId: PropTypes.string,
	};
  
	const navigation = useNavigation();
	return (
		<Button
			buttonStyle={styles.buttons}
			color='invisible'
			radius='20'
			size='sm'
			onPress={() =>
				navigation.navigate(navigationPlace, { courseId: courseId })
			}
			icon={
				<Icon
					size={25}
					name='chevron-left'
					type='material-community'
					color={projectColors.projectBlack}
				/>
			}
		></Button>
	);
};
const styles = StyleSheet.create({
	buttons: { width: 50, height: 50 },
});

export default LeaveButton;
