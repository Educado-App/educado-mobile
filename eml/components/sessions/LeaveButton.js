import React from 'react'
import { StyleSheet, TouchableOpacity } from 'react-native'
import { Icon, Button } from '@rneui/base'
const LeaveButton = () => {
<<<<<<< HEAD

    return (
      <TouchableOpacity>
          <Button buttonStyle={styles.buttons} color='white' radius='20' size='sm' icon={
          <Icon
              size={30}
              name='chevron-left'
              type='material-community'
              color='black'
            />}>
          </Button>
      </TouchableOpacity>
)
};
=======
  return (
    <TouchableOpacity>
      <Button
        buttonStyle={styles.buttons}
        color="invisible"
        radius="20"
        size="sm"
        icon={
          <Icon
            size={40}
            name="chevron-left"
            type="material-community"
            color="black"
          />
        }
      ></Button>
    </TouchableOpacity>
  )
}
>>>>>>> 0e61f0a1c43888d401629d05d780853b7d6f1208
const styles = StyleSheet.create({
  buttons: { width: 50, height: 50 }
})

export default LeaveButton
