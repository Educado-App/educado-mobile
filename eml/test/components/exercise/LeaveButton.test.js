import React from 'react';
import renderer from 'react-test-renderer';
import LeaveButton from '../../../components/exercise/LeaveButton';

let navigated = false;

jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({
    navigate: jest.fn(() => {
      navigated = true;
    }),
  }),
  useRoute: () => ({
    params: {},
  }),
}));

let leaveButtonScreen;

describe('LeaveButton', () => {
  it('renders correctly', async () => {
    await renderer.act(async () => {
      return (leaveButtonScreen = renderer.create(
        <LeaveButton navigationPlace='examplePlace' courseId={123} />
      ));
    });

    await new Promise((r) => setTimeout(r, 0));

    expect(leaveButtonScreen.toJSON()).toMatchSnapshot();
  });
});
