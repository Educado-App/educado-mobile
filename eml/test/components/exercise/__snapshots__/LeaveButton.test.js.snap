// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaveButton renders correctly 1`] = `
<View
  style={
    [
      {
        "overflow": "hidden",
      },
      {
        "borderRadius": 20,
      },
      undefined,
      false,
    ]
  }
  testID="RNE_BUTTON_WRAPPER"
>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": false,
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "invisible",
          "borderColor": "#2089dc",
          "borderRadius": 20,
          "borderWidth": 0,
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "center",
          "padding": 4,
          "paddingHorizontal": 6,
          "width": 50,
        }
      }
    >
      <View
        style={
          {
            "overflow": "hidden",
          }
        }
        testID="RNE__ICON__CONTAINER"
      >
        <View
          accessibilityRole="button"
          android_ripple={
            {
              "borderless": false,
              "color": "rgba(0, 0, 0, 0.3)",
              "radius": -5,
            }
          }
          disabled={false}
          testID="RNE__ICON__CONTAINER_ACTION"
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
            testID="RNE__ICON"
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#383838",
                    "fontSize": 25,
                  },
                  {
                    "backgroundColor": "transparent",
                  },
                  {
                    "fontFamily": "material-community",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
              testID="RNE__ICON__Component"
            >
              Û∞ÖÅ
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
